WordPress/src/main/java/org/wordpress/android/ui/posts/RemotePreviewLogicHelper.kt

```diff
@@ -0,0 +1,103 @@
+package org.wordpress.android.ui.posts
+
+import android.app.Activity
+import org.wordpress.android.fluxc.model.PostModel
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.ui.WPWebViewActivity
+import org.wordpress.android.fluxc.model.post.PostStatus
+import org.wordpress.android.ui.ActivityLauncherWrapper
+import org.wordpress.android.util.NetworkUtilsWrapper
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class RemotePreviewLogicHelper @Inject constructor(
+    private val networkUtilsWrapper: NetworkUtilsWrapper,
+    private val activityLauncherWrapper: ActivityLauncherWrapper
+) {
+    enum class RemotePreviewType {
+        NOT_A_REMOTE_PREVIEW,
+        REMOTE_PREVIEW,
+        REMOTE_PREVIEW_WITH_REMOTE_AUTO_SAVE
+    }
+
+    enum class PreviewLogicOperationResult {
+        PREVIEW_NOT_AVAILABLE,
+        NETWORK_NOT_AVAILABLE,
+        MEDIA_UPLOAD_IN_PROGRESS,
+        CANNOT_SAVE_EMPTY_DRAFT,
+        GENERATING_PREVIEW,
+        OPENING_PREVIEW,
+        CANNOT_REMOTE_AUTO_SAVE_EMPTY_POST
+    }
+
+    interface RemotePreviewHelperFunctions {
+        fun isNewPost() = false
```

Comment:
We should remove `isNewPost()` from the preview logic. `post.isLocalDraft` will always be true when `isNewPost` is true. It seems that `isNewPost` was introduced to show some dialog/info in the editor and do some analytics tracking. Unfortunately other devs later used it for publishing logic and it was probably a mistake.

Reply:
Understand the rational and thanks for sharing the background, it seems obvious I didn't get it fully 😊. Makes sense to remove isNewPost and canDiscard as you suggest in this next comment. Changes here e8e97532281aa2adce1528bce409a30f303dff00

Reply:
Changes looks good in e8e9753